components/DatePicker.js

```diff
@@ -0,0 +1,236 @@
+    calendarContainer.classList.add('bx--date-picker__calendar');
+    calendarContainer.querySelector('.flatpickr-month').classList.add('bx--date-picker__month');
+    calendarContainer.querySelector('.flatpickr-weekdays').classList.add('bx--date-picker__weekdays');
+    calendarContainer.querySelector('.flatpickr-days').classList.add('bx--date-picker__days');
+    [...calendarContainer.querySelectorAll('.flatpickr-weekday')].forEach((item) => {
+      const currentItem = item;
+      currentItem.innerHTML = currentItem.innerHTML.replace(/\s+/g, '');
+      currentItem.classList.add('bx--date-picker__weekday');
+    });
+    [...calendarContainer.querySelectorAll('.flatpickr-day')].forEach((item) => {
+      item.classList.add('bx--date-picker__day');
+      if (item.classList.contains('today') && calendar.selectedDates.length > 0) {
+        item.classList.add('no-border');
+      } else if (item.classList.contains('today') && calendar.selectedDates.length === 0) {
+        item.classList.remove('no-border');
+      }
+    });
+  }
+
+  updateInputFields = (selectedDates) => {
+    const input = ReactDOM.findDOMNode(this.refs.inputField).querySelector('.bx--date-picker__input');
+    if (this.props.datePickerType === 'range') {
+      const toInput = ReactDOM.findDOMNode(this.refs.toInputField).querySelector('.bx--date-picker__input');
+      if (selectedDates.length === 2) {
+        input.value = this.formatDate(selectedDates[0]);
+        toInput.value = this.formatDate(selectedDates[1]);
+      } else if (selectedDates.length === 1) {
+        input.value = this.formatDate(selectedDates[0]);
+      }
+    } else if (selectedDates.length === 1) {
+      input.value = this.formatDate(selectedDates[0]);
+    }
+    this.updateClassNames(this.state.cal);
+  }
+
+  formatDate = date => this.state.cal.formatDate(date, this.props.dateFormat);
+
+  render() {
+    const {
+      children,
+      className,
+      short,
+      datePickerType,
+      dateFormat, // eslint-disable-line
+      ...other
+    } = this.props;
+
+    const datePickerClasses = classNames('bx--date-picker', className, {
+      'bx--date-picker--short': short,
+      'bx--date-picker--simple': datePickerType === 'simple',
+      'bx--date-picker--single': datePickerType === 'single',
+      'bx--date-picker--range': datePickerType === 'range',
+    })
+
+    const datePickerIcon = (datePickerType === 'range')
+    ? (<svg onClick={this.openCalendar} className="bx--date-picker__icon" width="17" height="19" viewBox="0 0 17 19">
+        <path d="M12 0h2v2.7h-2zM3 0h2v2.7H3z"/>
+        <path d="M0 2v17h17V2H0zm15 15H2V7h13v10z"/>
+        <path d="M9.9 15H8.6v-3.9H7.1v-.9c.9 0 1.7-.3 1.8-1.2h1v6z"/>
+      </svg>)
+    : '';
+
+    const childArray = React.Children.toArray(children);
+    const childrenWithProps =
+      childArray.map((child, index) => {
+        if (index === 0) {
+          return React.cloneElement(child, {
+            datePickerType,
```

Comment:
It doesn't look like you need to include `datePickerType` explicitly like this. When the user declares this on `<DatePicker>`, then it still works. 

Removing this will also prevent this warning from popping up:

```
  console.error node_modules/fbjs/lib/warning.js:36
    Warning: Unknown prop `datePickerType` on <div> tag. Remove this prop from the element. For details, see https://fb.me/react-unknown-prop
        in div (created by DatePicker)
        in div (created by DatePicker)
        in div (created by DatePicker)
        in DatePicker (created by WrapperComponent)
        in WrapperComponent
```